File: example/src/main/java/com/github/vivchar/example/widgets/AvatarView.java
package com.github.vivchar.example.widgets;

import android.content.Context;
import android.util.AttributeSet;

import androidx.annotation.NonNull;
import androidx.appcompat.widget.AppCompatImageView;

import com.bumptech.glide.Glide;
import com.bumptech.glide.load.MultiTransformation;
import com.bumptech.glide.load.resource.bitmap.CircleCrop;
import com.bumptech.glide.request.RequestOptions;

import jp.wasabeef.glide.transformations.BlurTransformation;

/**
 * Created by Vivchar Vitaly on 10.07.19.
 */

public class AvatarView extends AppCompatImageView {

	public AvatarView(final Context context) {
		this(context, null);
	}

	public AvatarView(final Context context, final AttributeSet attrs) {
		this(context, attrs, 0);
	}

	public AvatarView(final Context context, final AttributeSet attrs, final int defStyleAttr) {
		super(context, attrs, defStyleAttr);
	}

	//Temporary workaround, you should use view's attr to circle view
	public void setUrl(@NonNull final String url, final boolean isCircle) {
		final RequestOptions options = new RequestOptions();
		if (isCircle) {
			options.transform(new MultiTransformation(new BlurTransformation(25), new CircleCrop()));
		} else {
			options.transform(new BlurTransformation(25));
		}

		Glide.with(getContext())
				.asBitmap()
				.load(url)
				.apply(options)
				.into(this);
	}

	public void setUrl(@NonNull final String url) {
		setUrl(url, false);
	}
}

File: example/src/main/java/com/github/vivchar/example/widgets/AvatarView.kt
package com.github.vivchar.example.widgets

import android.content.Context
import android.util.AttributeSet
import androidx.appcompat.widget.AppCompatImageView
import com.bumptech.glide.Glide
import com.bumptech.glide.load.MultiTransformation
import com.bumptech.glide.load.resource.bitmap.CircleCrop
import com.bumptech.glide.request.RequestOptions
import jp.wasabeef.glide.transformations.BlurTransformation

/**
 * Created by Vivchar Vitaly on 10.07.19.
 */
class AvatarView @JvmOverloads constructor(
	context: Context,
	attrs: AttributeSet? = null,
	defStyleAttr: Int = 0
) : AppCompatImageView(context, attrs, defStyleAttr) {

	//Temporary workaround, you should use view's attr to circle view
	fun setUrl(url: String, isCircle: Boolean) {
		RequestOptions().let {
			it.transform(
				if (isCircle) MultiTransformation(BlurTransformation(25), CircleCrop())
				else BlurTransformation(25)
			)
			Glide.with(context)
				.asBitmap()
				.load(url)
				.apply(it)
				.into(this)
		}
	}

	fun setUrl(url: String) = setUrl(url, false)
}